### ISSUE TITLE
Add brief & detail mode

### ISSUE THREAD

**Issue by bsbrother** [leptonai/search_with_lepton#53]
https://github.com/vlinx-io/infinite-search
Compared with Search With Lepton, we provide two information processing flows.

* Brief Mode
    Search keywords with search engine APIs and obtain search results.
    Aggregate the snippets of the search results as context and integrate it into the prompt.
    Submit the prompt to the LLM and obtain the results.
* Detail Mode
   Search keywords with search engine APIs and obtain search results.
   Store the webpage content of the top two search results and the snippets of the remaining search results in a vector database.
   Retrieve the most similar texts from the vector database and integrate them into the prompt as contextual information.
   Submit the prompt to the LLM and obtain the results.

Can add these functions? thank you!

**Comment by kyiree** [comment]
i do not think this is a good idea , as a smart AI search engine, it should give the right answer even use the Brief Mode

**Comment by kyiree** [comment]
and the top two search results is not that precise, why not top three results ?
